text/1003-cli-naming.rst

```diff
@@ -19,110 +19,238 @@ specified.
-such as ``--password-from-stdin``.
+Command-line design doesn't really follow EdgeQL
+------------------------------------------------
 
-Summarizing the above:
+**Concern**: The initial design goal of using the ``<action>-<category>``
+scheme was to have the CLI naming follow EdgeQL commands, e.g.
+``$ edgedb create-superuser-role`` and ``CREATE SUPERUSER ROLE`` look
+similar. However this similarity is only superficial as command-line tools
+have additional CLI-specific options, such as ``--password-from-stdin``.
 
-1. With more and more functionality exposed to the CLI, the
-   ``<action>-<category>`` approach has the potential to balloon the output
-   of ``edgedb -h`` to more than 100 lines, making it effectively unreadable.
+**Decision**: This is true.  One-to-one mapping between command-line
+functionality and EdgeQL should not be assumed by the user.  There isn't
+much we can realistically do to bridge this gap because modes of use
+are different between EdgeQL and a command-line tool:
 
-   The ``<group> <command>`` scheme would only display the few top-level
-   categories making it easier for users to build a mental model of what's
-   possible with the CLI.
+* different escaping rules allowing for more flexibility within EdgeQL
+  queries;
 
-2. The ``edgedb show-status`` command does not indicate what kind of status
-   it will show. It could be showing status of the current migration, or of
-   the currently running server, or of how many active connections the server
-   has at the moment.  Renaming it to ``edgedb show-migration-status`` would
-   make it longer to type and overall very verbose.
+* a multi-line editor for EdgeQL queries is provided but user shells
+  might not provide multi-line editing of terminal commands; and
 
-   With ``<group> <command>`` scheme the solution is obvious:
-   ``edgedb migration status``.
+* different information passing paradigms, with command-line tools using
+  standard I/O pipelines whereas EdgeQL using sub-queries.
 
-3. The ``edgedb show-status`` command prompts another question: what
-   if another kind of "display status" command is needed?
+Vague verbs or nouns can be misleading
+--------------------------------------
 
-   With ``<group> <command>`` scheme, there would be no conflict between
-   ``edgedb server status`` and ``edgedb migration status``.
+**Concern**: the ``edgedb show-status`` command does not indicate what kind
+of status it will show. It could be showing status of the current migration,
+or of the currently running server, or of how many active connections the
+server has at the moment. Renaming it to ``edgedb show-migration-status``
+would make it longer to type and overall very verbose. With
+``<group> <command>`` scheme the solution is obvious:
+``edgedb migration status``.
 
-4. The ``<action>-<category>`` scheme does not allow category-specific
-   help sections.
+**Decision**: ``edgedb show-status`` should be renamed to
```

Comment:
ACTIONABLE ITEM 2: rename `show-status` to `migration-status`.

Reply:
+1.